### ISSUE TITLE
gem-maven-plugin downloads for each version of the gem a pom file

### ISSUE THREAD

**Issue by drdee** [torquebox/jruby-maven-plugins#57]
Hi,

When I add a gem dependency to my JRuby project it downloads a pom file for each version of the gem. I am not entirely sure whether this is correct behavior or configurable behavior but it does slow down significantly fetching a dependency. Is this tweakable? If not , why is this happening?
(I am specifying explicit versions for each gem in the pom file)

I also get the following error when running mvn package:

``` shell
[ERROR] Failed to execute goal de.saumya.mojo:gem-maven-plugin:1.0.2:exec (default) on project longboat: error in executing jruby: loading resource from classloader failed: META-INF/jruby.home/bin/gem -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal de.saumya.mojo:gem-maven-plugin:1.0.2:exec (default) on project longboat: error in executing jruby
Caused by: java.io.FileNotFoundException: loading resource from classloader failed: META-INF/jruby.home/bin/gem
    at de.saumya.mojo.ruby.ScriptUtils.getScriptFromResource(ScriptUtils.java:47)
    at de.saumya.mojo.ruby.script.ScriptFactory.newScriptFromResource(ScriptFactory.java:150)
    at de.saumya.mojo.ruby.script.ScriptFactory.newScriptFromJRubyJar(ScriptFactory.java:115)
    at de.saumya.mojo.ruby.gems.GemsInstaller.maybeAddArtifact(GemsInstaller.java:230)
    at de.saumya.mojo.ruby.gems.GemsInstaller.installGems(GemsInstaller.java:143)
    at de.saumya.mojo.ruby.gems.GemsInstaller.installGems(GemsInstaller.java:114)
    at de.saumya.mojo.ruby.gems.GemsInstaller.installGems(GemsInstaller.java:102)
    at de.saumya.mojo.ruby.gems.GemsInstaller.installPom(GemsInstaller.java:61)
    at de.saumya.mojo.gem.AbstractGemMojo.executeJRuby(AbstractGemMojo.java:339)
    at de.saumya.mojo.jruby.AbstractJRubyMojo.execute(AbstractJRubyMojo.java:291)
```

This is using jruby-1.7.12 and version 1.0.2 of the gem-maven-plugin. The configuration of the gem-maven-plugin is as follows:

``` xml

 <plugin>
    <groupId>de.saumya.mojo</groupId>
    <artifactId>gem-maven-plugin</artifactId>
    <version>${jruby.plugins.version}</version>
    <executions>
        <execution>
            <goals>
                <goal>exec</goal>
            </goals>
            <phase>compile</phase>
            <configuration>
                <execArgs>bundle exec rake test</execArgs>
            </configuration>
        </execution>
    </executions>
    <extensions>true</extensions>
    <configuration>
        <includeRubygemsInResources>true</includeRubygemsInResources>
        <includeRubygemsInTestResources>false</includeRubygemsInTestResources>
    </configuration>
</plugin>
```

Best,
Diederik

**Comment by abelsromero** [comment]
Sorry @mkristian  to re-open the issue, but I had some troubles with this and I'd like to your opinion.
Trying to use 'asciidoctor-bibtext' as a maven dependency crashes because one of the dependency ranges tries to pull a gem that does not exist anymore. But newer versions solve this issue.
```
rubygems:namae:gem:0.8.2 -> rubygems:rubysl:gem:2.0.8 -> rubygems:rubysl-json:gem:[2.0,2.99999]: No versions available for rubygems:rubysl-json:gem:[2.0,2.99999] within specified range 
```

Could it be possible to skip this error so that if in the end a valid and newer version exists the plugin continues? or as a workaround, could it be possible to use the sets goal to override some of the dependencies instead of having to define all gems?

**Comment by mkristian** [comment]
@abelsromero the problem is really under the hood. the moment declare ```namae-0.8.2``` in your pom.xml then maven will resolve it before it hits any plugin or extension. ```namae-0.8.2``` is also particular in the sense that ```namae-0.8.1``` and ```namae-0.8.3``` do not have any dependency to rubysl - only version 0.8.2 does have this dependency. so I would ask the namae people to yank 0.8.2 as it clearly can not be installed anymore and gives errors with maven and gradle.

in short a plugin in maven can not fix this problem with empty version ranges after a project yanked all its gems.

but it is possible to fix this with exclusions:
```
gem 'namae', '0.8.2', :exclusions => ['rubygems:rubysl']
gem 'rubysl', '2.1.0'
```

**Comment by abelsromero** [comment]
So, I understand that the resolution is 100% delegated to maven and nothing can be done...well, was worth it a try :+1: 
In anycas I got it to work setting all gems manually using the set goal, but it's really a lot of work.
